pkg/util/modules/modules.go

```diff
@@ -25,6 +25,12 @@ type module struct {
 // in the right order of dependencies.
 type Manager struct {
 	modules map[string]*module
+
+	// Modules that are already initialized
+	modulesInitialized map[string]bool
```

Comment:
Let’s not introduce more state to the Manager than necessary. In particular `modulesInitialized` and `servicesMap` don’t need to be fields of the manager.

Calling `InitModuleServices` twice should recreate everything from scratch.

Reply:
Done